packages/clarity-js/src/data/extract.ts

```diff
@@ -67,6 +88,19 @@ export function compute(): void {
                 }
             }
         }
+        for (let h in hashes) {
+            let key = parseInt(h);
+            let hashData = hashes[key];
+            for (let d in hashData) {
+                let hashKey = parseInt(d);
+                let content = hashText(hashData[hashKey]);
+                // only update when the contents have changes since we last checked
+                if (content != hashDataCache[hashData[hashKey]]) {
+                    hashDataCache[hashData[hashKey]] = content;
+                    update(key, hashKey, content.substring(0, Setting.ExtractLimit));
```

Comment:
instead of maintaining a seperate dictionary - pass in a additional parameter like force which if true would update if value has changed in data dictionary